environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldSEBeanRegistrant.java

```diff
@@ -203,4 +214,181 @@ public boolean isNullable() {
         }
 
     }
+
+    void observesPATofDeprecatedActivator(@Observes @WithAnnotations(ActivateThreadScope.class) ProcessAnnotatedType<?> event) {
+        final AnnotatedType<?> originalAT = event.getAnnotatedType();
+        WeldSELogger.LOG.deprecatedActivatorAnnotationUsed(ActivateThreadScope.class.getName(),
+                org.jboss.weld.environment.se.contexts.activators.ActivateThreadScope.class.getName());
+        event.setAnnotatedType(new AnnotatedTypeWrapper(originalAT));
+    }
 }
+
+class AnnotatedTypeWrapper<X> implements AnnotatedType<X> {
+
+    private final AnnotatedType<X> delegate;
+    private final Set<AnnotatedMethod<? super X>> annotatedMethods = new HashSet<>();
+    private final Set<Annotation> annotations = new HashSet<>();
+
+    public AnnotatedTypeWrapper(AnnotatedType<X> annotatedType) {
+        this.delegate = annotatedType;
+        replaceDeprecatedAnnotatedMethods();
```

Comment:
A more clear approach is to return an initialized reference and set it to a final field, i.e.:
```
this.annotatedMethods = wrapAnnotatedMethods(delegate.getMethods());
```
Also the set should be immutable.